spec.html

```diff
@@ -37705,8 +37707,8 @@ <h1>ValidateAtomicAccess ( _typedArray_, _requestIndex_ )</h1>
         <p>The abstract operation ValidateAtomicAccess takes arguments _typedArray_ and _requestIndex_. It performs the following steps when called:</p>
         <emu-alg>
           1. Assert: _typedArray_ is an Object that has a [[ViewedArrayBuffer]] internal slot.
-          1. Let _accessIndex_ be ? ToIndex(_requestIndex_).
           1. Let _length_ be _typedArray_.[[ArrayLength]].
+          1. Let _accessIndex_ be ? ToIndex(_requestIndex_).
```

Comment:
is there any reason to flip these? it wouldn't be observable either way, ofc.

Reply:
This is what @anba and I have been talking about in the thread. The invariant is that [[ArrayLength]] shouldn't be accessed on a TA whose buffer is detached, which ToIndex can cause to happen. One benefit of doing that is an implementation can implement getting a TA's length as returning 0 when its buffer is detached without thinking hard about if it is observable.

The reordering is strictly speaking not observable but the intent is clearer to have it come first, to lessen the chances a naive implementation is wrong.